CONTRIBUTING.md

```diff
@@ -12,29 +9,104 @@ You can contribute to Turing in several ways:
 - Create articles and documentation for users and contributors
 - Help others answering questions about Turing
 
-### Report bugs
+The following sections provide some suggestions and guidelines you should follow:
+- [Code of Conduct](#coc)
+- [Issues and Bugs](#issue)
+- [Feature Requests](#feature)
+- [Submission Guidelines](#submit)
+- [Coding Rules](#rules)
+
+## <a name="coc"></a> Code of Conduct
+
+Help us keep Turing open and inclusive.
+Please read and follow our [Code of Conduct](https://github.com/caraml-dev/turing/blob/main/CODE_OF_CONDUCT.md).
+
+## <a name="issue"></a> Found a Bug?
+
+If you find a bug in the source code, you can help us by [submitting an issue](#submit-issue) to our [GitHub Repository][github].
+Even better, you can [submit a Pull Request](#submit-pr) with a fix.
+
+
+## <a name="feature"></a> Missing a Feature?
+You can *request* a new feature by [submitting an issue](#submit-issue) to our GitHub Repository.
+If you would like to *implement* a new feature, please consider the size of the change in order to determine the right steps to proceed:
+
+* For a **Major Feature**, first open an issue and outline your proposal so that it can be discussed.
+  This process allows us to better coordinate our efforts, prevent duplication of work, and help you to craft the change so that it is successfully accepted into the project.
+
+  **Note**: Adding a new topic to the documentation, or significantly re-writing a topic, counts as a major feature.
+
+* **Small Features** can be crafted and directly [submitted as a Pull Request](#submit-pr).
+
+## <a name="submit"></a> Submission Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
+
+Before you submit an issue, please search the issue tracker. An issue for your problem might already exist and the discussion might inform you of workarounds readily available.
+
+We want to fix all the issues as soon as possible, but before fixing a bug, we need to reproduce and confirm it.
+In order to reproduce bugs, we require that you provide a minimal reproduction.
+Having a minimal reproducible scenario gives us the necessary information without going back and forth to you with additional questions.
+
+A minimal reproduction allows us to quickly confirm a bug (or point out a coding problem) as well as confirm that we are fixing the right problem.
+
+We require a minimal reproduction to save maintainers' time and ultimately be able to fix more bugs.
+
+Unfortunately, we are not able to investigate / fix bugs without a minimal reproduction, so if we don't hear back from you, we are going to close an issue that doesn't have enough information to be reproduced.
+
+You can file new issues by selecting from our [new issue templates](https://github.com/caraml-dev/turing/issues/new/choose) and filling out the issue template.
+
+### <a name="submit-pr"></a> Submitting a Pull Request (PR)
+
+Before you submit your Pull Request (PR), consider the following guidelines:
```

Comment:
Does this mean that every PR should be accompanied by an issue, going forward?

Related q:
* Should we start tracking every feature / bug in GitHub issues?
* Should we have a PR template?

Reply:
>Should we start tracking every feature / bug in GitHub issues?

Ideally yes.

>Should we have a PR template?

Added!